diff -Npru protobuf-2.5.0.orig/src/google/protobuf/stubs/platform_macros.h protobuf-2.5.0/src/google/protobuf/stubs/platform_macros.h
--- protobuf-2.5.0.orig/src/google/protobuf/stubs/platform_macros.h	2014-01-11 16:59:44.019835679 -0500
+++ protobuf-2.5.0/src/google/protobuf/stubs/platform_macros.h	2014-01-11 17:11:57.697528481 -0500
@@ -52,6 +52,9 @@
 #elif defined(__ARMEL__)
 #define GOOGLE_PROTOBUF_ARCH_ARM 1
 #define GOOGLE_PROTOBUF_ARCH_32_BIT 1
+#elif defined(__mips__)
+#define GOOGLE_PROTOBUF_ARCH_MIPS 1
+#define GOOGLE_PROTOBUF_ARCH_32_BIT 1
 #elif defined(__MIPSEL__)
 #define GOOGLE_PROTOBUF_ARCH_MIPS 1
 #define GOOGLE_PROTOBUF_ARCH_32_BIT 1
